test.sh

```diff
@@ -1,80 +1,62 @@
+    echo "      \$AWS_ACCESS_KEY_ID"
+    echo "      \$AWS_SECRET_ACCESS_KEY"
+    echo "- Current frameworks:"
+    for framework in $FRAMEWORK_LIST; do
+        echo "       $framework"
+    done
+    exit 1
+fi
 
-        if [ ! -f "$UNIVERSE_URL_PATH" ]; then
-            echo "Missing universe URL file: $UNIVERSE_URL_PATH"
-            exit 1
-        fi
-        export STUB_UNIVERSE_URL=$(cat $UNIVERSE_URL_PATH)
-        rm -f $UNIVERSE_URL_PATH
-        echo "Built/uploaded stub universe: $STUB_UNIVERSE_URL"
-    else
-        echo "Using provided STUB_UNIVERSE_URL: $STUB_UNIVERSE_URL"
-    fi
+if [ -z "$CLUSTER_URL" ]; then
+    echo "Cluster not found. Create and configure one then set \$CLUSTER_URL."
+    exit 1
+fi
 
-    echo Security: $SECURITY
-    if [ "$SECURITY" = "strict" ]; then
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root ${framework}-role
-        # include foldered roles (tests exercise with /test/integration/svcname):
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root test__integration__${framework}-role
-    fi
+if [ -z "$AWS_ACCESS_KEY_ID" -o -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    echo "AWS credentials not found (\$AWS_ACCESS_KEY_ID and \$AWS_SECRET_ACCESS_KEY)."
+    exit 1
+fi
 
-    # Run shakedown tests in framework directory:
-    TEST_GITHUB_LABEL="${framework}" ${REPO_ROOT_DIR}/tools/run_tests.py shakedown ${FRAMEWORK_DIR}/tests/
-}
+if [ "$1" = "all" ]; then
+    # randomize the FRAMEWORK_LIST
+    FRAMEWORK_LIST=$(while read -r fw; do printf "%05d %s\n" "$RANDOM" "$fw"; done <<< "$FRAMEWORK_LIST" | sort -n | cut -c7- )
+else
+    FRAMEWORK_LIST=$1
+fi
 
 echo "Beginning integration tests at "`date`
 
-REPO_ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-
-cd $REPO_ROOT_DIR
+for framework in $FRAMEWORK_LIST; do
+    FRAMEWORK_DIR=${REPO_ROOT_DIR}/frameworks/${framework}
 
-# Get a CCM cluster if not already configured (see available settings in dcos-commons/tools/README.md):
-if [ -z "$CLUSTER_URL" ]; then
-    echo "CLUSTER_URL is empty/unset, launching new cluster."
-    export CCM_AGENTS=6
-    CLUSTER_INFO=$(${REPO_ROOT_DIR}/tools/launch_ccm_cluster.py)
-    echo "Launched cluster: ${CLUSTER_INFO}"
-    # jq emits json strings by default: "value".  Use --raw-output to get value without quotes
-    export CLUSTER_URL=$(echo "${CLUSTER_INFO}" | jq --raw-output .url)
-    export CLUSTER_ID=$(echo "${CLUSTER_INFO}" | jq .id)
-    export CLUSTER_AUTH_TOKEN=$(echo "${CLUSTER_INFO}" | jq --raw-output .auth_token)
-    CLUSTER_CREATED="true"
-else
-    echo "Using provided CLUSTER_URL as cluster: $CLUSTER_URL"
-    CLUSTER_CREATED=""
-fi
+    echo "Starting build for $framework at "`date`
+    export UNIVERSE_URL_PATH=${FRAMEWORK_DIR}/$framework-universe-url
+    ${FRAMEWORK_DIR}/build.sh aws
+    if [ ! -f "$UNIVERSE_URL_PATH" ]; then
+        echo "Missing universe URL file: $UNIVERSE_URL_PATH"
+        exit 1
+    fi
+    export STUB_UNIVERSE_URL=$(cat $UNIVERSE_URL_PATH)
```

Comment:
I took that block of logic out because I wasn't sure how to handle it for the case where all the frameworks were being tested. Would there just be one STUB_UNIVERSE_URL for them all? Or would one need to be defined for each separately? As soon as it seemed like there would be more complex modes to the script I opted for simplicity.